File: wikidox_server/Dto/UserDataDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WikiDoxServer.Dto
{
    public class UserDataDto
    {
        public string session { get; set; } = "";
        public string username { get; set; } = "";
        public string password { get; set; } = "";
        public string salt { get; set; } ="";
        public int canEdit { get; set; } = 0;
        public int canUpload { get; set; } = 0;
        public int canCreate { get; set; } = 0;
        public int isAdmin { get; set; } = 0;
    }
}

File: wikidox_client/src/app/dto/user-data-dto.ts
export class UserDataDto {
    session="";
    username="";
    password="";
    canEdit=0;
    canCreate=0;
    canUpload=0;
    isAdmin=0;
    constructor(session,username,password,canCreate=false,canEdit=false, canUpload=false, isAdmin=false)
    {
	this.session=session;
	this.username=username;
	this.password=password;
	this.canEdit=canEdit?1:0;
	this.canCreate=canCreate?1:0;
	this.canUpload=canUpload?1:0;
	this.isAdmin=isAdmin?1:0;
    }
}
